test/implementations/functions_testing.py

```diff
@@ -576,90 +557,88 @@ def verify_circuit_dirty_wasted_relative_separable_auxiliary(
-        inverse_matrix = load_matrix("reverse_noauxiliary", controls_no)
 
-        b_shape = 2 ** (controls_no + 1)
-        c_shape = 2 ** (auxiliaries_no)
+    # get mct inverse matrix
+    inverse_matrix = load_matrix("reverse_noauxiliary", controls_no)
 
-        b, c = reverse_kronecker_product(unitary_matrix, (b_shape, b_shape))
+    b_shape = 2 ** (controls_no + 1)
+    c_shape = 2 ** (auxiliaries_no)
 
-        no_of_qubits = controls_no + 1
+    b, c = reverse_kronecker_product(unitary_matrix, (b_shape, b_shape))
 
-        # check shape of c
-        if c.shape != (c_shape, c_shape):
-            return False, "Unitary C has different shape."
+    no_of_qubits = controls_no + 1
 
-        # check if c is unitary
-        check_c = np.matmul(c, np.linalg.inv(c))
+    # check shape of c
+    if c.shape != (c_shape, c_shape):
+        return False, "Unitary C has different shape."
 
-        if (
-            np.allclose(
-                check_c,
-                identity_matrix(auxiliaries_no),
-                atol=absolute_error_tol,
-                rtol=relative_error_tol,
-            )
-            is False
-        ):
-            return False, "Something wrong with the implementation"
+    # check if c is unitary
+    check_c = np.matmul(c, np.linalg.inv(c))
 
-        # Expected unitary after calculation is identity.
-        expected_unitary = identity_matrix(no_of_qubits)
+    if (
+        np.allclose(
+            check_c,
+            identity_matrix(auxiliaries_no),
+            atol=absolute_error_tol,
+            rtol=relative_error_tol,
+        )
+        is False
+    ):
+        return False, "Something wrong with the implementation"
 
-        # X_1 * X_2^dagger = I
-        m = np.matmul(b, inverse_matrix)
-        generated_unitary = np.absolute(m)
+    # Expected unitary after calculation is identity.
+    expected_unitary = identity_matrix(no_of_qubits)
 
-        if generated_unitary.shape != expected_unitary.shape:
-            return False, "Unitary B has different shape."
+    # X_1 * X_2^dagger = I
+    m = np.matmul(b, inverse_matrix)
+    generated_unitary = np.absolute(m)
 
-        if (
-            np.allclose(
-                generated_unitary,
-                expected_unitary,
-                atol=absolute_error_tol,
-                rtol=relative_error_tol,
-            )
-            is False
-        ):
-            return False, "Something wrong with the implementation"
+    if generated_unitary.shape != expected_unitary.shape:
+        return False, "Unitary B has different shape."
+
+    if (
+        np.allclose(
+            generated_unitary,
+            expected_unitary,
+            atol=absolute_error_tol,
+            rtol=relative_error_tol,
+        )
+        is False
+    ):
+        return False, "Something wrong with the implementation"
 
-        return True, ""
-    except Exception:
-        return False, ""
+    return True, ""
 
 
-def verify_dependencies(rd):

```

Comment:
why it is commented out? I'm not saying it is bad, but I want to understand

Reply:
i put it in each of the implementations, test_dependencies.

Also, I will fix this with another issue with the recursivel.. that is why I commented them out.

Reply:
will be fixed in issue #15